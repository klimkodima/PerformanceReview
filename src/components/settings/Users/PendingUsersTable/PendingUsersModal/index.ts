export { default as PendingUsersModal } from './PendingUsersModal';
