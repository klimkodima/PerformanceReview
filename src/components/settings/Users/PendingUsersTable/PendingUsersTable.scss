.pending_users {
  .MuiPaper-root {
    overflow-y: hidden;
  }
  button {
    margin-top: 30px;
  }
  button.pending_users__stop-saving-button {
    margin-left: 25px;
  }
}

.pending-users-modal {
  max-width: 85vw;
  width: 100%;
  h2 {
    font-size: 22px;
    padding: 0 0 20px;
    color: #777777;
    text-align: center;
  }
  &__table-container {
    padding: 25px 0 45px;
    td {
      border: none;
    }
    td {
      svg {
        margin-right: 10px;
        vertical-align: middle;
      }
    }
    td:nth-of-type(1) {
      font-weight: 700;
      color: #525252;
    }
  }
  .MuiFormControl-root {
    width: 160px;
  }
  select {
    padding-top: 4px;
    padding-bottom: 4px;
  }
  .css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root {
    top: -12px;
  }
  td.pending-users-modal__icon-block {
    display: flex;
  }
}
