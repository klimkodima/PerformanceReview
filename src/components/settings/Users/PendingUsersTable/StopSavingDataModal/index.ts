export { default as StopSavingDataModal } from './StopSavingDataModal';
