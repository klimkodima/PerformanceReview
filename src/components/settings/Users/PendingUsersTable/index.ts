export { default as PendingUsersTable } from './PendingUsersTable';
