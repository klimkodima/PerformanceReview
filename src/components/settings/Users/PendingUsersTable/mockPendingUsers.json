[
  {
    "id": 5,
    "fullName": "Pety Petrov",
    "username": "Pety Petrov",
    "email": "p.petrov@gpsolutions.com",
    "enabled": true,
    "roleName": "",
    "pending": true,
    "jiraAccountId": "12866hg",
    "worksFrom": "08-31-2015",
    "thirdParty": false
  },
  {
    "id": 6,
    "fullName": "Egor Egorov",
    "username": "Egor Egorov",
    "email": "e.egorov@gpsolutions.com",
    "enabled": false,
    "roleName": "",
    "teamId": 1,
    "teamName": "X-Rays",
    "level": "TEAM_LEAD",
    "pending": true,
    "jiraAccountId": "12866hg",
    "worksFrom": "08-31-2015",
    "thirdParty": false
  }
]
