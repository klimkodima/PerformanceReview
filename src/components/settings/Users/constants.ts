import { t } from 'i18next';

export const HEADER_CELLS_PENDING_USERS: string[] = [
  t('users_table.user_name_label'),
  t('users_table.email_label'),
  t('users_table.level_label'),
  t('users_table.team_label'),
  t('pending_users_table.data_storage_label')
];

export const TABLE_PENDING_TITLE = t('users_table.table_pending_title');

export const ADMIN_ACCESS_LEVEL = t('access_level.admin');

export const USERS_ROW_CELLS: string[] = [
  'fullName',
  'email',
  'enabled',
  'roleName',
  'level',
  'teamName'
];

export const PENDING_ROW_CELLS: string[] = [
  'fullName',
  'email',
  'level',
  'teamName'
];

export const HEADER_CELLS_USERS: HeaderCellsUserType[] = [
  { name: t('users_table.user_name_label'), isShowIcon: false },
  { name: t('users_table.email_label'), isShowIcon: false },
  { name: t('users_table.status_label'), isShowIcon: false },
  { name: t('users_table.role_label'), isShowIcon: true },
  { name: t('users_table.level_label'), isShowIcon: true },
  { name: t('users_table.team_label'), isShowIcon: true }
];

export type HeaderCellsUserType = {
  name: string;
  isShowIcon: boolean;
};

export const TEAM_LABEL = t('users_table.team_label');
export const ROLE_LABEL = t('users_table.role_label');
export const LEVEL_LABEL = t('users_table.level_label');

export const ADMIN_ROLE = 'ADMIN';
export const MANAGER_ROLE = 'MANAGER';
export const TEAM_LEAD_ROLE = 'TEAM_LEADER';
export const AUDITOR_ROLE = 'AUDITOR';

export const TEAM_LEAD_LEVEL = 'TEAM_LEAD';

export const ENABLED_KEY = 'enabled';
export const ROLE_NAME_KEY = 'roleName';
export const LEVEL_KEY = 'level';

export const ACTIVATE_BUTTON = t('pending_users_table.activate_button');
export const STOP_SAVING_DATA_BUTTON = t(
  'pending_users_table.stop_saving_data_button'
);

export const CANCEL_BUTTON = t('pending_users_table.cancel_button');
export const MODAL_TITLE = t('pending_users_table.title');
export const MODAL_SUBTITLE = t('pending_users_table.subtitle');
export const MODAL_PRESET_TITLE = t('pending_users_table.modal_preset_title');
export const MODAL_INFO_TOOLTIP_TEXT = t(
  'pending_users_table.modal_info_tooltip_text'
);
export const SELECT_LABEL = t('pending_users_table.select_label');
export const PRESET_WORKING_HOURS_KEY = 'thirdParty';

export const OPTION_1 = t('pending_users_table.option_role_1');
export const OPTION_2 = t('pending_users_table.option_role_2');
export const OPTION_3 = t('pending_users_table.option_role_3');
export const OPTION_4 = t('pending_users_table.option_role_4');

export enum Roles {
  ADMIN = 1,
  MANAGER = 2,
  TEAM_LEADER = 3,
  AUDITOR = 4
}

export enum Levels {
  TEAM_LEAD = 1,
  SENIOR = 2,
  MIDDLE = 3,
  JUNIOR = 4,
  TRAINEE = 5,
  NULL = 6
}
