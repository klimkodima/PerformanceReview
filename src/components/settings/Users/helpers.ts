import { SelectChangeEvent } from '@mui/material';
import { ChangeEvent } from 'react';

import {
  AUDITOR_ROLE,
  PRESET_WORKING_HOURS_KEY,
  TEAM_LEAD_LEVEL,
  TEAM_LEAD_ROLE
} from './constants';

import { UserInfoType } from 'src/store/Users/types';
import { UpdatedPendingUsersType } from './types';

export const onChangePendingUsers = (
  e: SelectChangeEvent<string> | ChangeEvent<HTMLInputElement>,
  idRow: number,
  rowsState: UpdatedPendingUsersType[],
  setRowsState: (arg: UpdatedPendingUsersType[]) => void
): void => {
  const { value, name } = e.target;

  const newRowsState = rowsState.map((row) => {
    if (row.id === idRow) {
      if (name === PRESET_WORKING_HOURS_KEY) {
        return {
          ...row,
          [name]: !row.thirdParty
        };
      }
      return {
        ...row,
        [name]: value
      };
    }
    return row;
  });

  setRowsState(newRowsState);
};

export const getSelectedUsersInfo = (
  selectedUsers: UserInfoType[],
  setUsersSelectedInfo: (value: UpdatedPendingUsersType[]) => void
): void => {
  const usersSelectedInfo: UpdatedPendingUsersType[] = [];
  selectedUsers.map(({ id, thirdParty, level }: UserInfoType) => {
    let role = AUDITOR_ROLE;
    if (level === TEAM_LEAD_LEVEL) role = TEAM_LEAD_ROLE;
    if (!level) role = '';

    usersSelectedInfo.push({
      id,
      thirdParty,
      role
    });
  });

  setUsersSelectedInfo(usersSelectedInfo);
};

export const getNamesString = (
  selectedUsers: UserInfoType[],
  isEnabled: boolean
): string => {
  let usersString = '';
  selectedUsers.map(({ fullName, enabled }: UserInfoType) => {
    const searchValue = isEnabled ? enabled : !enabled;
    return searchValue ? (usersString += `, ${fullName}`) : '';
  });
  return usersString.slice(1);
};
