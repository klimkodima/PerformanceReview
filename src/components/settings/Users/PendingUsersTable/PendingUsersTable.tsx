import { ReactElement, useState, useEffect } from 'react';

import { isEmpty } from 'lodash';
import { useDispatch, useSelector } from 'react-redux';
import {
  TableContainer,
  Paper,
  Table,
  TableHead,
  TableRow,
  TableCell,
  TableBody,
  Checkbox,
  Button
} from '@mui/material';

import { createFormattedString } from 'src/utils';
import {
  activateUser,
  fetchUsers,
  selectUsers,
  stopSavingData
} from 'src/store/Users';
import { selectSettingsPermission } from 'src/store/App';
import { SETTINGS_PERMISSION } from 'src/constants';
import { PendingUsersModal } from './PendingUsersModal';
import { StopSavingDataModal } from './StopSavingDataModal';
import {
  ACTIVATE_BUTTON,
  HEADER_CELLS_PENDING_USERS,
  PENDING_ROW_CELLS,
  STOP_SAVING_DATA_BUTTON,
  TABLE_PENDING_TITLE
} from '../constants';
import { getSelectedUsersInfo } from '../helpers';
import { handleCheckboxClick } from '../../common/helpers';

import { UserInfoType } from 'src/store/Users/types';
import { UpdatedPendingUsersType } from '../types';

import './PendingUsersTable.scss';

const PendingUsersTable = (): ReactElement => {
  const settingsPermission = useSelector(selectSettingsPermission);
  const usersData = useSelector(selectUsers);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchUsers());
  }, []);

  const copyUsersData: UserInfoType[] = JSON.parse(JSON.stringify(usersData));
  const pendingUsers = copyUsersData.filter(({ pending }) => pending);

  const isEdit = settingsPermission === SETTINGS_PERMISSION.WRITE;
  const editClassName = isEdit ? 'settings-table__edit' : '';

  const [selected, setSelected] = useState<string[]>([]);
  const [selectedUsers, setSelectedUsers] = useState<UserInfoType[]>([]);
  const isDisabled = isEmpty(selected);
  const [isDisabledStopSavingButton, setIsDisabledStopSavingButton] =
    useState<boolean>(true);

  useEffect(() => {
    const checkedUsers = pendingUsers.filter(({ id }) =>
      selected.includes(id.toString())
    );
    setSelectedUsers(checkedUsers);

    const saveDataUsers: boolean = isEmpty(
      checkedUsers.filter(({ enabled }) => enabled)
    );
    setIsDisabledStopSavingButton(saveDataUsers);
  }, [selected]);

  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
  const [isStopSavingModalOpen, setIsStopSavingModalOpen] =
    useState<boolean>(false);
  const handleModalOpenClick = (
    setIsModalOpen: (value: boolean) => void
  ): void => {
    setIsModalOpen(true);
  };
  const handleModalCloseClick = (
    setIsModalOpen: (value: boolean) => void
  ): void => {
    setIsModalOpen(false);
    setSelected([]);
  };

  useEffect(() => {
    getSelectedUsersInfo(selectedUsers, setUsersSelectedInfo);
  }, [selectedUsers]);

  const [usersSelectedInfo, setUsersSelectedInfo] = useState<
    UpdatedPendingUsersType[]
  >([]);

  const [isDisabledSendBtn, setIsDisabledSendBtn] = useState<boolean>(true);
  useEffect(() => {
    const withoutRole = usersSelectedInfo.filter(({ role }) => !role);
    isEmpty(withoutRole)
      ? setIsDisabledSendBtn(false)
      : setIsDisabledSendBtn(true);
  }, [usersSelectedInfo]);

  const activatePendingUsers = () => {
    dispatch(activateUser(usersSelectedInfo));
    handleModalCloseClick(setIsModalOpen);
  };

  const stopSavingDataForUsers = () => {
    dispatch(stopSavingData(selectedUsers));
    handleModalCloseClick(setIsStopSavingModalOpen);
  };

  useEffect(() => {
    if (isModalOpen || isStopSavingModalOpen)
      document.body.classList.add('scroll');
    return () => {
      document.body.classList.remove('scroll');
    };
  }, [isModalOpen, isStopSavingModalOpen]);

  return (
    <div className='pending_users'>
      <h2>{TABLE_PENDING_TITLE}</h2>
      <div className='settings-table'>
        <div className='settings-table__wrapper'>
          <TableContainer component={Paper} className={editClassName}>
            <Table aria-label='table'>
              <TableHead>
                <TableRow>
                  {isEdit && <TableCell />}
                  {HEADER_CELLS_PENDING_USERS.map((headCell) => (
                    <TableCell key={headCell}>{headCell}</TableCell>
                  ))}
                </TableRow>
              </TableHead>
              <TableBody>
                {pendingUsers.map((user: UserInfoType) => (
                  <TableRow key={user.id}>
                    {isEdit && (
                      <TableCell>
                        <Checkbox
                          checked={selected.includes(user.id.toString())}
                          className='edit__checkbox'
                          inputProps={{ 'aria-label': 'checkbox' }}
                          color='success'
                          onClick={() =>
                            handleCheckboxClick(
                              user.id.toString(),
                              selected,
                              setSelected
                            )
                          }
                        />
                      </TableCell>
                    )}
                    {PENDING_ROW_CELLS.map((cell: string) => (
                      <TableCell key={cell}>
                        {user[cell]
                          ? createFormattedString(user[cell] as string)
                          : '-'}
                      </TableCell>
                    ))}
                    <TableCell>{user.enabled ? 'yes' : 'no'}</TableCell>
                  </TableRow>
                ))}
              </TableBody>
            </Table>
          </TableContainer>
          {isEdit && (
            <>
              <Button
                variant='outlined'
                color='success'
                disabled={isDisabled}
                onClick={() => handleModalOpenClick(setIsModalOpen)}
              >
                {ACTIVATE_BUTTON}
              </Button>
              <Button
                variant='outlined'
                color='success'
                className='pending_users__stop-saving-button'
                disabled={isDisabledStopSavingButton}
                onClick={() => handleModalOpenClick(setIsStopSavingModalOpen)}
              >
                {STOP_SAVING_DATA_BUTTON}
              </Button>
            </>
          )}
        </div>
      </div>
      {isEdit && isModalOpen && (
        <PendingUsersModal
          selectedUsers={selectedUsers}
          usersSelectedInfo={usersSelectedInfo}
          setUsersSelectedInfo={setUsersSelectedInfo}
          onCloseClick={() => handleModalCloseClick(setIsModalOpen)}
          onSaveClick={activatePendingUsers}
          isDisabledSendBtn={isDisabledSendBtn}
        />
      )}
      {isEdit && isStopSavingModalOpen && (
        <StopSavingDataModal
          onCloseClick={() => handleModalCloseClick(setIsStopSavingModalOpen)}
          selectedUsers={selectedUsers}
          onSaveClick={stopSavingDataForUsers}
        />
      )}
    </div>
  );
};

export default PendingUsersTable;
