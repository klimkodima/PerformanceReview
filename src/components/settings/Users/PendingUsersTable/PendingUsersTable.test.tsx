import { I18nextProvider } from 'react-i18next';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Provider } from 'react-redux';

import store from 'src/store';
import i18n from 'src/i18n';
import { setUsers } from 'src/store/Users';
import { setAvailableWidgets } from 'src/store/App';
import mockUsersData from 'src/store/Users/mockUsers.json';
import PendingUsersTable from './PendingUsersTable';

beforeEach(() => {
  store.dispatch(setUsers(mockUsersData));
  store.dispatch(
    setAvailableWidgets({
      settingsPermission: 'WRITE',
      availableWidgets: []
    })
  );
});

describe('Pending Users table tests', () => {
  test('Pending Users table snapshot', () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <I18nextProvider i18n={i18n}>
            <PendingUsersTable />
          </I18nextProvider>
        </Provider>
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  test('Pending Users table has the elements', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'portal');
    document.body.appendChild(root);

    const { getByRole, getAllByRole } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <PendingUsersTable />
        </I18nextProvider>
      </Provider>
    );

    expect(getByRole('table')).toBeInTheDocument();
    expect(getAllByRole('columnheader')).toHaveLength(6);
    expect(getAllByRole('row')).toHaveLength(3);

    const activateUsersBtn = getByRole('button', { name: 'Activate Users' });
    expect(activateUsersBtn).toBeInTheDocument();
    expect(activateUsersBtn).toHaveProperty('disabled', true);
    const checkboxes = getAllByRole('checkbox');
    expect(checkboxes).toHaveLength(2);

    userEvent.click(checkboxes[0]);
    userEvent.click(checkboxes[1]);
    expect(activateUsersBtn).toHaveProperty('disabled', false);
    userEvent.click(activateUsersBtn);
    expect(getAllByRole('heading', { level: 2 })).toHaveLength(2);
    expect(getAllByRole('row')).toHaveLength(5);
    expect(getAllByRole('button')).toHaveLength(4);

    userEvent.click(getByRole('button', { name: 'Cancel' }));
    expect(getAllByRole('row')).toHaveLength(3);
    expect(activateUsersBtn).toHaveProperty('disabled', true);
  });

  test('Pending Users Modal activate users', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'portal');
    document.body.appendChild(root);

    const { getByRole, getAllByRole, getAllByTestId, queryByTestId } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <PendingUsersTable />
        </I18nextProvider>
      </Provider>
    );

    const activateUsersBtn = getByRole('button', { name: 'Activate Users' });
    const checkboxes = getAllByRole('checkbox');
    userEvent.click(checkboxes[1]);
    userEvent.click(activateUsersBtn);

    expect(getAllByRole('row')).toHaveLength(4);
    const modalCheckboxes = getAllByTestId('pending-users-modal-checkbox');
    expect(modalCheckboxes).toHaveLength(1);

    const modalActivateBtn = getAllByRole('button', {
      name: 'Activate Users'
    });
    expect(modalActivateBtn).toHaveLength(2);
    const selects = getAllByTestId('pending-users-modal-select');
    expect(selects).toHaveLength(1);
    expect(modalActivateBtn[1]).toHaveProperty('disabled', false);
    userEvent.click(modalCheckboxes[0]);
    userEvent.click(modalActivateBtn[0]);
    expect(getAllByRole('row')).toHaveLength(3);
    expect(queryByTestId('modal-overlay')).toBeNull();
  });

  test('Pending Users Modal stop saving data for users', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'portal');
    document.body.appendChild(root);

    const { getByRole, getAllByRole, getByTestId, queryByTestId } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <PendingUsersTable />
        </I18nextProvider>
      </Provider>
    );

    const stopSavingBtn = getByRole('button', {
      name: 'Stop saving data for Users'
    });
    expect(stopSavingBtn).toHaveProperty('disabled', true);
    const checkboxes = getAllByRole('checkbox');
    userEvent.click(checkboxes[0]);
    expect(stopSavingBtn).toHaveProperty('disabled', true);
    userEvent.click(checkboxes[1]);
    expect(stopSavingBtn).toHaveProperty('disabled', false);
    userEvent.click(stopSavingBtn);
    expect(getByTestId('modal-overlay')).toBeInTheDocument();
    expect(getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
    userEvent.click(getByRole('button', { name: 'Yes' }));
    expect(queryByTestId('modal-overlay')).toBeNull();
  });

  test('Pending Users Modal is closing', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'portal');
    document.body.appendChild(root);

    const { getByRole, getAllByRole, getByTestId, queryByTestId } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <PendingUsersTable />
        </I18nextProvider>
      </Provider>
    );

    const stopSavingBtn = getByRole('button', {
      name: 'Stop saving data for Users'
    });
    const checkboxes = getAllByRole('checkbox');
    userEvent.click(checkboxes[0]);
    userEvent.click(checkboxes[1]);
    userEvent.click(stopSavingBtn);

    expect(getByTestId('modal-overlay')).toBeInTheDocument();
    userEvent.click(getByRole('button', { name: 'Cancel' }));
    expect(queryByTestId('modal-overlay')).toBeNull();
  });
});
