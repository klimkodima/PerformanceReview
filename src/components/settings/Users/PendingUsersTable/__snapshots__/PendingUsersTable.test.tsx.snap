// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pending Users table tests Pending Users table snapshot 1`] = `
<div
  className="pending_users"
>
  <h2>
    Pending users
  </h2>
  <div
    className="settings-table"
  >
    <div
      className="settings-table__wrapper"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root settings-table__edit css-11xur9t-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          aria-label="table"
          className="MuiTable-root css-rqglhn-MuiTable-root"
          role={null}
        >
          <thead
            className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              />
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                User Name
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Email
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Level
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Team
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Data storage
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <span
                  className="MuiCheckbox-root MuiCheckbox-colorSuccess MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSuccess PrivateSwitchBase-root edit__checkbox css-10drtbx-MuiButtonBase-root-MuiCheckbox-root"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <input
                    aria-label="checkbox"
                    checked={false}
                    className="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckBoxOutlineBlankIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Pety petrov
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                p.petrov@gpsolutions.com
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                -
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                -
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                no
              </td>
            </tr>
            <tr
              className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <span
                  className="MuiCheckbox-root MuiCheckbox-colorSuccess MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSuccess PrivateSwitchBase-root edit__checkbox css-10drtbx-MuiButtonBase-root-MuiCheckbox-root"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <input
                    aria-label="checkbox"
                    checked={false}
                    className="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckBoxOutlineBlankIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Egor egorov
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                e.egorov@gpsolutions.com
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Junior
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                X-rays
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                yes
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <button
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-sxix9q-MuiButtonBase-root-MuiButton-root"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        Activate Users
      </button>
      <button
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled pending_users__stop-saving-button css-sxix9q-MuiButtonBase-root-MuiButton-root"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        Stop saving data for Users
      </button>
    </div>
  </div>
</div>
`;
